src/lib/storage/reference_segment.cpp

```diff
@@ -6,38 +6,39 @@
 namespace opossum {
 
 ReferenceSegment::ReferenceSegment(const std::shared_ptr<const Table>& referenced_table,
-                                   const ColumnID referenced_column_id, const std::shared_ptr<const PosList>& pos) {
-  // Implementation goes here
-}
+                                   const ColumnID referenced_column_id, const std::shared_ptr<const PosList>& pos)
+    : _referenced_table{referenced_table}, _referenced_column_id{referenced_column_id}, _pos_list{pos} {}
 
 AllTypeVariant ReferenceSegment::operator[](const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  const auto row_id = _pos_list->at(chunk_offset);
```

Comment:
I would suggest an assertion before whether the given chunk_offset can be called in the pos list? 
```suggestion
  Assert(chunk_offset < _pos_list->size(), "Invalid offset given for chunk;
  const auto row_id = _pos_list->at(chunk_offset);
```